sign_mac.sh

```diff
@@ -6,28 +6,49 @@ set -x
 echo "***** MAC SIGNING *****"
 
 PKG_DIR=obj/pkg
+SIGN_ARGS=(-s "$MAC_CERT_NAME" -fv --requirement "=designated => ( (anchor apple generic and certificate leaf[field.1.2.840.113635.100.6.1.9] ) or (anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] and certificate leaf[field.1.2.840.113635.100.6.1.13] and certificate leaf[subject.OU] = \"${MAC_CERT_NAME}\"))" -o "runtime" --entitlements "mozilla-release/security/mac/hardenedruntime/production.entitlements.xml")
 
 for DMG in obj/dist/*.dmg
 do
   echo "Processing $DMG..."
   rm -f -rf $PKG_DIR
   mkdir $PKG_DIR
-  hdiutil attach -nobrowse $DMG
-  cp -r /Volumes/cliqz/*.app $PKG_DIR
-  xattr -rc $PKG_DIR/
-  hdiutil detach /Volumes/cliqz
-  for app in $PKG_DIR/*.app
-  do
-    codesign -s $MAC_CERT_NAME --force --deep $app
-
-    # copy back to dist folder a signed app (for generating an update package(s) later)
-    if [[ $DMG == *"de.mac.dmg"* ]]; then
-      cp -r $app obj/dist/l10n-stage/cliqz
-    fi
-    if [[ $DMG == *"en-US.mac.dmg"* ]]; then
-      cp -r $app obj/dist/cliqz
-    fi
-  done
+  mozilla-release/build/package/mac_osx/unpack-diskimage $DMG /Volumes/Cliqz $PKG_DIR
+  xattr -cr $PKG_DIR/Cliqz.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/plugin-container.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/crashreporter.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/updater.app
+
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/plugin-container.app
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/pingsender
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libfreebl3.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/cliqz-bin
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/liblgpllibs.dylib
```

Comment:
Could we codesign all `dylibs` with a glob pattern, wildcard or for loop?

Reply:
I'm not good in bash scripting